### PR TITLE
Extract method in SelfSigned

### PR DESCRIPTION
**Pull request by ignasi35**
This is a noop change. extracts a method in preparation for a refactor that will eventually move this code to ssl-config.

### CODE REVIEW

**Comment by ignasi35** [comment]
The plan is to move the extracted function away so that we can build `SSLContext` (et al.) given _any_ `keystore`. I think this could eventually be part of a new `testkit` module in https://github.com/lightbend/ssl-config

There is quite a few places/projects that have very similar needs (fake certs, trust stores built on-the-fly,...).

The long term effort would solve https://github.com/akka/akka-grpc/issues/89

**Comment by renatocaval** [comment]
Travis said yes!

File: play-testkit/src/main/scala/play/api/test/SelfSigned.scala [hunk 1]
```diff
@@ -24,16 +24,21 @@ object SelfSigned {
...
+  private def buildContextAndTrust(keyStore: KeyStore) = {
 
-    val tmf: TrustManagerFactory = TrustManagerFactory
-      .getInstance(TrustManagerFactory.getDefaultAlgorithm())
+    val tmf: TrustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm)
     tmf.init(keyStore)
     val tms: Array[TrustManager] = tmf.getTrustManagers
     val x509TrustManager: X509TrustManager = tms(0).asInstanceOf[X509TrustManager]
 
+    val kmf: KeyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm)
```
Comment (dwijnand) [inline play-testkit/src/main/scala/play/api/test/SelfSigned.scala]: @ignasi35 was the change from "SunX509" to `KeyManagerFactory.getDefaultAlgorithm` intentional?

Comment (dwijnand) [reply play-testkit/src/main/scala/play/api/test/SelfSigned.scala]: Moreover, is it significant?

Comment (ignasi35) [reply play-testkit/src/main/scala/play/api/test/SelfSigned.scala]: It was intentional and _I think_ it's not significant.

**Pull request merged**